concent_api/core/tests/test_auth_get_task_result.py

```diff
@@ -0,0 +1,964 @@
+
+from core.models            import MessageAuth
+from core.tests.utils       import ConcentIntegrationTestCase
+from utils.testing_helpers  import generate_ecc_key_pair
+
+
+(CONCENT_PRIVATE_KEY, CONCENT_PUBLIC_KEY)                         = generate_ecc_key_pair()
+(DIFFERENT_PROVIDER_PRIVATE_KEY, DIFFERENT_PROVIDER_PUBLIC_KEY)   = generate_ecc_key_pair()
+(DIFFERENT_REQUESTOR_PRIVATE_KEY, DIFFERENT_REQUESTOR_PUBLIC_KEY) = generate_ecc_key_pair()
+
+
+def get_file_status_true_mock(_):
+    return True
+
+
+def get_file_status_false_mock(_):
+    return False
+
+
+@override_settings(
+    CONCENT_PRIVATE_KEY    = CONCENT_PRIVATE_KEY,
+    CONCENT_PUBLIC_KEY     = CONCENT_PUBLIC_KEY,
+    CONCENT_MESSAGING_TIME = 10,  # seconds
+    FORCE_ACCEPTANCE_TIME  = 10,  # seconds
+)
+class GetTaskResultIntegrationTest(ConcentIntegrationTestCase):
+
+    def test_requestor_forces_get_task_result_and_concent_immediately_sends_rejection_due_to_already_sent_message_should_work_only_with_correct_keys(self):
+        """
+        Tests if on requestor ForceGetTaskResult message Concent will return ForceGetTaskResultRejected
+        if ForceGetTaskResult was already sent by this requestor.
+
+        Expected message exchange:
+        Requestor               -> Concent:                 ForceGetTaskResult
+        Concent                 -> Requestor:               ForceGetTaskResultAck
+        WrongRequestor/Provider -> Concent                  ForceGetTaskResult
+        Concent                 -> WrongRequestor/Provider  ForceGetTaskResultAck
+        Requestor               -> Concent:                 ForceGetTaskResult
+        Concent                 -> Requestor:               ForceGetTaskResultRejected
+        """
+
+        # STEP 1: Requestor forces get task result via Concent.
+        deserialized_task_to_compute = self._get_deserialized_task_to_compute(
+            timestamp = "2017-12-01 10:00:00",
+            deadline  = "2017-12-01 11:00:00",
+        )
+        deserialized_report_computed_task = self._get_deserialized_report_computed_task(
+            task_to_compute = deserialized_task_to_compute
+        )
+        original_serialized_force_get_task_result = self._get_serialized_force_get_task_result(
+            report_computed_task = deserialized_report_computed_task,
+            timestamp            = "2017-12-01 11:00:08",
+        )
+
+        with freeze_time("2017-12-01 11:00:08"):
+            response = self.client.post(
+                reverse('core:send'),
+                data                                = original_serialized_force_get_task_result,
+                content_type                        = 'application/octet-stream',
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY      = self._get_encoded_requestor_public_key(),
+                HTTP_CONCENT_OTHER_PARTY_PUBLIC_KEY = self._get_encoded_provider_public_key(),
+            )
+
+        assert response.status_code == 200
```

Comment:
This is part of the test so it should be `assertEqual()`. If it's not (i.e. if you consider it a step that prepares data for the test and not the actual test) then checks below should use `assert` too.

Reply:
:+1: